server/src/graphql/queries/repositories.ts

```diff
@@ -0,0 +1,100 @@
+import fetch from 'node-fetch';
+import { QueryResolvers } from '../../generated/graphql';
+import jsonwebtoken from 'jsonwebtoken';
+import { config } from '../../config';
+import { prisma } from '../../prisma';
+export const repositories: QueryResolvers['repositories'] = async (
+  _,
+  { installationId },
+  { userId }
+) => {
+  if (!userId) {
+    throw new Error('Unauthorized');
+  }
+
+  const user = await prisma.user.findUnique({
+    where: {
+      id: userId,
+    },
+  });
+
+  const payload = {
+    iat: Math.round(new Date().getTime() / 1000),
+    exp: Math.round(new Date().getTime() / 1000 + 60),
+    iss:
+      process.env.NODE_ENV === 'production'
+        ? config.githubAppId
+        : process.env.GITHUB_APP__ID,
+  };
+
+  //TODO FIGURE OUT A WAY TO PARSE THE PEM PROPERLY
+  const token = jsonwebtoken.sign(
+    payload,
+    `-----BEGIN RSA PRIVATE KEY-----
+---- KEY
+-----END RSA PRIVATE KEY-----`,
+    {
+      algorithm: 'RS256',
+    }
+  );
+  const authAsInstallation = await fetch(
+    `https://api.github.com/app/installations/${installationId}/access_tokens`,
+    {
+      headers: {
+        Accept: 'application/vnd.github.machine-man-preview+json',
+        Authorization: `Bearer ${token}`,
+      },
+      method: 'POST',
+    }
+  );
+  const data = await authAsInstallation.json();
+  console.log(data);
+  const responseRepos = await fetch(
+    'https://api.github.com/installation/repositories',
+    {
+      headers: {
+        Accept: 'application/vnd.github.v3+json',
+        Authorization: `token ${data.token}`,
+      },
+    }
+  );
+
+  const fetchedRepos = await responseRepos.json();
+
+  let repositories = [];
+  let branches = [];
+
+  const map = fetchedRepos.repositories.map(async (r) => {
```

Comment:
so here you are fetching all the branches for all the repos, if 100 repos are returned it will result in 100 HTTP requests made. I think we should have 2 queries: this one which is good and exposes another one that will accept a github repo id and return the branches for this repo.
On the frontend every time the user switch the repo using the select input we would call the branches query and load the values for only one repo. That way this query will resolve much faster :)